jack-mixer 17 fails to build with Cython 3

I am trying to build jack-mixer on Fedora 39 with python 3.12.
The build fails with the following error:
```
/usr/bin/cython -o src/_jack_mixer.c ../src/_jack_mixer.pyx

Error compiling Cython file:
------------------------------------------------------------
...
        self._midi_change_callback = callback
        if callback is None:
            channel_set_midi_change_callback(self._channel, NULL, NULL)
        else:
            channel_set_midi_change_callback(self._channel,
                                             &midi_change_callback_func,
                                             ^
------------------------------------------------------------

/builddir/build/BUILD/jack_mixer-release-17/src/_jack_mixer.pyx:352:45: Cannot assign type 'void (*)(void *) except * nogil' to 'midi_change_callback'. Exception values are incompatible. Suggest adding 'noexcept' to type 'void (void *) except * nogil'.
[10/13] gcc -Isrc/jack_mix_box.p -Isrc -I../src -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include -I/usr/include/sysprof-6 -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wextra -O2 -flto=auto -ffat-lto-objects -fexceptions -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-U_FORTIFY_SOURCE,-D_FORTIFY_SOURCE=3 -Wp,-D_GLIBCXX_ASSERTIONS -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1 -m64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer -Wno-error=format-security -pthread '-DLOCALEDIR="/usr/share/locale"' -DHAVE_JACK_MIDI -MD -MQ src/jack_mix_box.p/jack_mixer.c.o -MF src/jack_mix_box.p/jack_mixer.c.o.d -o src/jack_mix_box.p/jack_mixer.c.o -c ../src/jack_mixer.c
ninja: build stopped: subcommand failed.
I
```

--- a/README.md
+++ b/README.md
@@ -17,7 +17,7 @@ information.
 To build and install jack_mixer run:
 
 ```console
-meson builddir --prefix=/usr --buildtype=release
+meson setup builddir --prefix=/usr --buildtype=release
 meson compile -C builddir
 [sudo] meson install -C builddir
 ```

--- a/src/_jack_mixer.pxd
+++ b/src/_jack_mixer.pxd
@@ -25,7 +25,7 @@ cdef extern from "jack_mixer.h":
     ctypedef void * jack_mixer_output_channel_t
     ctypedef void * jack_mixer_threshold_t
 
-    ctypedef void (*midi_change_callback)(void *)
+    ctypedef void (*midi_change_callback)(void *) noexcept with gil
 
     cdef enum midi_behavior_mode:
         pass

--- a/src/_jack_mixer.pyx
+++ b/src/_jack_mixer.pyx
@@ -10,7 +10,7 @@ import enum
 from _jack_mixer cimport *
 
 
-cdef void midi_change_callback_func(void *userdata) with gil:
+cdef void midi_change_callback_func(void *userdata) noexcept with gil:
     """Wrapper for a Python callback function for MIDI input."""
     channel = <object> userdata
     channel._midi_change_callback()

--- a/src/meson.build
+++ b/src/meson.build
@@ -35,7 +35,7 @@ if get_option('gui').enabled()
     jack_mixer_mod_pyx = '_jack_mixer.pyx'
     jack_mixer_mod_c = '_jack_mixer.c'
 
-    cython = find_program('cython3', 'cython', required: false)
+    cython = find_program('cython3', 'cython', 'cython0', required: false)
 
     if fs.exists(jack_mixer_mod_c)
         jack_mixer_cython = files(jack_mixer_mod_c)

